### PR TITLE
feat: use Nest.js SDK in backend

### PR DESCRIPTION
**Pull request by lukas-reining**
## This PR

Adds the Nest.js SDK to the backend services.

### CODE REVIEW

File: packages/app/src/app/app.module.ts [hunk 1]
```diff
@@ -1,50 +1,26 @@
 import {HttpModule} from '@nestjs/axios';
-import {MiddlewareConsumer, Module, NestModule, Scope} from '@nestjs/common';
-import {REQUEST} from '@nestjs/core';
+import {ExecutionContext, Module} from '@nestjs/common';
 import {ServeStaticModule} from '@nestjs/serve-static';
-import {AsyncLocalStorageTransactionContext, LoggingHook, OpenFeatureLogger} from '@openfeature/extra';
-import { FlagMetadata, OpenFeature } from '@openfeature/server-sdk';
-import {MetricsHook, TracingHook as SpanEventBasedTracingHook} from '@openfeature/open-telemetry-hooks';
+import {LoggingHook, OpenFeatureLogger} from '@openfeature/extra';
+import {FlagMetadata} from '@openfeature/js-sdk';
```
Comment (beeme1mr) [inline packages/app/src/app/app.module.ts:5]: I'm surprised this worked unless we didn't remove the `js-sdk` from thepackage.json. This should be exported from the nest sdk, please use that one.

File: packages/app/src/app/fibonacci/fibonacci.service.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
-import { HttpService } from '@nestjs/axios';
-import { Inject, Injectable } from '@nestjs/common';
-import { fibonacci } from '@openfeature/fibonacci';
-import { Client } from '@openfeature/server-sdk';
-import { OPENFEATURE_CLIENT } from '../constants';
-import { lastValueFrom, map } from 'rxjs';
+import {HttpService} from '@nestjs/axios';
+import {Injectable} from '@nestjs/common';
+import {fibonacci} from '@openfeature/fibonacci';
+import {Client} from '@openfeature/js-sdk';
```
Comment (beeme1mr) [inline packages/app/src/app/fibonacci/fibonacci.service.ts:4]: Please use the nest sdk

File: packages/fibonacci-service/src/app/app.module.ts [hunk 3]
```diff
@@ -1,40 +1,19 @@
-import {MiddlewareConsumer, Module, NestModule} from '@nestjs/common';
+import {Module} from '@nestjs/common';
 import {AppController} from './app.controller';
 import {LoggerModule} from 'nestjs-pino';
-import { FlagMetadata, OpenFeature } from '@openfeature/server-sdk';
-import {AsyncLocalStorageTransactionContext, LoggingHook, OpenFeatureLogger} from '@openfeature/extra';
+import {FlagMetadata} from '@openfeature/server-sdk';
```
Comment (beeme1mr) [inline packages/fibonacci-service/src/app/app.module.ts:4]: Please use `@openfeature/nestjs-sdk`.

**Review by beeme1mr — COMMENTED** [review]
Nice, looks good. Were you able to run this locally? I'll try and test it out this afternoon.

**Comment by lukas-reining** [comment]
Yes @beeme1mr, building it locally and running worked!
I will change the imports tomorrow.

**Comment by lukas-reining** [comment]
I updated the imports @toddbaert @beeme1mr :)

**Review by beeme1mr — COMMENTED** [review]
I'm running into an issue building the UI. It seems to have something to do with missing webpack polyfills. I'm looking into it.

**Review by beeme1mr — APPROVED** [review]

**Comment by beeme1mr** [comment]
> I'm running into an issue building the UI. It seems to have something to do with missing webpack polyfills. I'm looking into it.

I found the issue. There was a shared library that was used both on the client and server but was importing from the Nest SDK. I've updated the imports and added a linting rule to prevent it from happening in the future.

**Comment by lukas-reining** [comment]
> > I'm running into an issue building the UI. It seems to have something to do with missing webpack polyfills. I'm looking into it.
> 
> 
> 
> I found the issue. There was a shared library that was used both on the client and server but was importing from the Nest SDK. I've updated the imports and added a linting rule to prevent it from happening in the future.

Mh interesting, I could build it with the former `npm run temp` and thought I cleaned everything before 🤔
Thank you for fixing it!

**Pull request merged**